pkg/controller/astarte/reconcile/utils.go

```diff
@@ -390,10 +390,15 @@ func getImageForClusteredResource(defaultImageName, defaultImageTag string, reso
 	return image
 }
 
-func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource) appsv1.DeploymentStrategy {
+func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource, component apiv1alpha1.AstarteComponent) appsv1.DeploymentStrategy {
 	if resource.DeploymentStrategy != nil {
 		return *resource.DeploymentStrategy
 	}
+	if component.DashedString() == "data-updater-plant" || component.DashedString() == "flow" {
```

Comment:
1. Don't use literals, use the correct constants
2. This is wrong on several levels, `cr.Spec.DeploymentStrategy` should become a pointer, then:

* if it's `nil`, then this logic is correct and you should return Recreate or RollingUpdate
* if it's not nil, you should return its value

If you don't do this, you're overriding the user's choices